### PR TITLE
Use getfullargspec instead of getargspec (where available) to silence deprecation warning.

### LINKED ISSUE #1515: Config calls inspect.getargspec() which is deprecated, resulting in a warning
The warning is as follows:

```
  /Users/riceb2/.local/share/virtualenvs/guvnor-iF69-5_x/lib/python3.6/site-packages/gunicorn/config.py:417: DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
    if arity != -1 and arity != len(inspect.getargspec(val)[0]):
```

### PR DESCRIPTION
**Pull request by jeamland**
Fixes #1515.

### CODE REVIEW

File: gunicorn/config.py [hunk 1]
```diff
@@ -396,6 +396,12 @@ def validate_class(val):
     return validate_string(val)
 
 
+if hasattr(inspect, 'getfullargspec'):
```
Comment (berkerpeksag) [inline gunicorn/config.py]: `getattr(inspect, 'getfullargspec', inspect.getargspec)` would be shorter.

Comment (jeamland) [inline gunicorn/config.py]: Doing that now.

**Comment by jeamland** [comment]
Are the lint tests failing on master too? They appear to be based on what I can see...

**Comment by berkerpeksag** [comment]
It looks like for some reason prospector ignored the `ignore-paths` setting at https://github.com/benoitc/gunicorn/blob/master/.prospector.yaml#L3

File: gunicorn/config.py [hunk 2]
```diff
@@ -396,6 +396,9 @@ def validate_class(val):
     return validate_string(val)
 
 
+getargspec = getattr(inspect, 'getfullargspec', inspect.getargspec)
```
Comment (berkerpeksag) [inline gunicorn/config.py]: Sorry, I forgot to say this in my earlier comment, but could you please move this to `gunicorn/_compat.py`?

Comment (jeamland) [inline gunicorn/config.py]: Done.

**Comment by jeamland** [comment]
lint is something to do with prospector 0.12.5. Locking the version to 0.12.4 in tox.ini makes the lint failures stop. I'm digging in to that now.

**Pull request closed without merging**